language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -630,6 +732,73 @@ private static class ArgListsHolder {
         private List<String> replaceFunctionCallArgs;
     }
 
+    static class ExpressionNodeValidator extends NodeVisitor {
```

Comment:
Let's introduce a class(separate java file) for this ExpressionNodeValidator class as this is used in other code actions.